src/runtime/step_runner.js

```diff
@@ -8,22 +8,30 @@ const {beginTiming, endTiming} = Time
 
 async function run({attachmentManager, defaultTimeout, scenarioResult, step, stepDefinition, parameterRegistry, world}) {
   beginTiming()
-  const parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
-  const timeoutInMilliseconds = stepDefinition.options.timeout || defaultTimeout
-
-  let error, result
-  const validCodeLengths = stepDefinition.getValidCodeLengths(parameters)
-  if (_.includes(validCodeLengths, stepDefinition.code.length)) {
-    const data = await UserCodeRunner.run({
-      argsArray: parameters,
-      fn: stepDefinition.code,
-      thisArg: world,
-      timeoutInMilliseconds
-    })
-    error = data.error
-    result = data.result
-  } else {
-    error = stepDefinition.getInvalidCodeLengthMessage(parameters)
+  let error, result, parameters
+
+  try {
+    parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
+  } catch(err) {
+    error = err
+  }
+
+  if(!error) {
```

Comment:
Readability is important. But even more important is to display appreciation for contributions. When you nitpick like this it can easily be interpreted as "I'm never going to live up to this guy's standards", and that turns contributors away.